### DISCUSSION TITLE
Headpose vectors scaling by 1.5

### DISCUSSION THREAD

**Discussion by saeidn95** [vladmandic/human#454] [Q&A]
I wonder where the factor 1.5 (https://github.com/vladmandic/human/blob/62396317f5c31b2ec5bc7de2e9eb53b8978b9f28/src/face/angles.ts#L97) in the size of the x and y coordinates come from.

**Comment by vladmandic** [comment]
honestly, don't remember exactly, been a while, but i believe its because box size is ~1.5x mesh size as a general item (although it can vary slightly).

**Reply by saeidn95** [reply]
Yes that makes perfect sense. What that x and y are scale to the size of the face.box (changes from one image to the next). What does make sense to me though is that z is scale by a fixed value of 128. As I understand that the scale of z is same is x.

**Reply by vladmandic** [reply]
it's not, that's bug in a facemesh model itself.
I've reported it to mediapipe long time ago, but they didn't do anything about it.

**Reply by saeidn95** [reply]
What do you mean by "It's not" Do you mean that there is bug in the network causing z not to be in the same  scale as x? and to fix that your solution  is to scale z by 128 separate from  x and y that you scale to box size?

**Reply by vladmandic** [reply]
yes, that's what I mean.

**Reply by saeidn95** [reply]
Very good! Mediapipe has a new version that might have overcome this bug. But it seems they no longer provide the model in the tflite format.

**Reply by saeidn95** [reply]
Here is my final understanding of how you account for the scaling bug in mediapipe model. Tracing through the whole code from the mesh detection all the way to calculateFaceAngle in line (https://github.com/vladmandic/human/blob/62396317f5c31b2ec5bc7de2e9eb53b8978b9f28/src/face/angles.ts#L101), the whole sequence of operations reduces to scaling x and y by a factor of ~1.5 to match the scale of z value; or alternatively scale z by (1/1.5) to match the scale of x, and y.
